test/integration/scheduler_perf/util.go

```diff
@@ -429,22 +429,21 @@ func (tc *throughputCollector) run(tCtx ktesting.TContext) {
 			// be scheduled immediately when the timer
 			// triggers. Instead we track the actual time stamps.
 			duration := now.Sub(lastSampleTime)
-			durationInSeconds := duration.Seconds()
-			throughput := float64(newScheduled) / durationInSeconds
 			expectedDuration := throughputSampleInterval * time.Duration(skipped+1)
 			errorMargin := (duration - expectedDuration).Seconds() / expectedDuration.Seconds() * 100
 			if tc.errorMargin > 0 && math.Abs(errorMargin) > tc.errorMargin {
 				// This might affect the result, report it.
-				tCtx.Errorf("ERROR: Expected throuput collector to sample at regular time intervals. The %d most recent intervals took %s instead of %s, a difference of %0.1f%%.", skipped+1, duration, expectedDuration, errorMargin)
+				klog.Infof("WARNING: Expected throughput collector to sample at regular time intervals. The %d most recent intervals took %s instead of %s, a difference of %0.1f%%.", skipped+1, duration, expectedDuration, errorMargin)
 			}
 
 			// To keep percentiles accurate, we have to record multiple samples with the same
 			// throughput value if we skipped some intervals.
+			throughput := float64(newScheduled) / duration.Seconds()
```

Comment:
Because we divide by the actual duration of the interval now, it is less critical that sampling happens regularly. But if it gets too bad, then result data quality can still suffer: dividing by a much smaller interval than expected makes that data point less reliable.

But I'm okay with turning the error into a warning. I'm still wondering about the root cause and whether there is a solution.

/lgtm

Reply:
> I'm still wondering about the root cause

My **assumption** is that the collecting time waves, something like the API request takes longer. Then one collection takes more than 1s, 1s300ms let's say, however, the ticks already arrived, on the next collection cycle, the realDuration would be 1s300ms, and it finishes on time, then next another collection cycle would have a duration of 700ms.

Reply:
Is the collector really doing API calls? I think it uses the informer cache.

Then the only reason for not getting scheduled is really just that the process was busy serving some other goroutines.

Reply:
That makes sense, we only manipulate the informer.